### PR TITLE
feat(credentials): drop API version prefix from the endpoint

### PR DESCRIPTION
**Pull request by David-Kristek**
Refs: IBM#160

---
## Status
**READY**

## Description
Remove version from api endpoint and handle it by sdk itself .

## Any special notes for your reviewer?

---

## Checklist
- [ ] Automated tests exist
- [ ] Updated Package Requirements (if required, and with maintainers' approval)
- [ ] Local unit tests performed
- [ ] Documentation exists [link]()
- [ ] Local pre-commit hooks performed
- [ ] Desired commit message set as PR title and description set above
- [ ] Link to relevant GitHub issue provided

### CODE REVIEW

File: src/genai/credentials.py [hunk 1]
```diff
@@ -19,3 +22,11 @@ def __init__(
         if api_endpoint is None:
             raise ValueError("api_endpoint must be provided")
         self.api_endpoint = api_endpoint
+        self.remove_version()
+
+    def remove_version(self) -> None:
+        parsed_url = parse_url(self.api_endpoint)
+
+        if parsed_url.path and "/v" in parsed_url.path:
```
Comment (Tomas2D) [inline src/genai/credentials.py:30]: This is an incorrect assumption. Some environments can be hidden behind some path. 
Just check if the path ends with `vX` (X = `numeric version`)

`re.match(r".\d$", api_endpoint)`

**Review by Tomas2D — CHANGES_REQUESTED** [review]

File: src/genai/credentials.py [hunk 2]
```diff
@@ -1,5 +1,8 @@
+from urllib3.util import parse_url
```
[the remark this answers is not in the document]
Comment (David-Kristek) [reply src/genai/credentials.py]: The library was already there, that's why I used it.

File: src/genai/credentials.py [hunk 3]
```diff
@@ -19,3 +22,10 @@ def __init__(
         if api_endpoint is None:
             raise ValueError("api_endpoint must be provided")
         self.api_endpoint = api_endpoint
+        self._remove_api_endpoint_version()
+
+    def _remove_api_endpoint_version(self) -> None:
+        has_version = re.search(r".\d$", self.api_endpoint)
+        if has_version:
+            self.api_endpoint = self.api_endpoint.rsplit("/v", 1)[0]
+            print("Warning: Api_endpoint should not contain any version, removing it")
```
Comment (Tomas2D) [inline src/genai/credentials.py:31]: use logger warning, not a basic print (look at `Model` class)

Comment (David-Kristek) [reply src/genai/credentials.py:31]: I thought the logging isnt displayed to the user?, there is:
`logging.getLogger(__name__).addHandler(logging.NullHandler())`

Comment (Tomas2D) [reply src/genai/credentials.py:31]:
```python
def _remove_api_endpoint_version(self) -> None:
    [api, *version] = re.split(r"(/v\d+$)", self.api_endpoint, maxsplit=1)
    if version:
        warn(
            DeprecationWarning(
                f"The 'api_endpoint' property should not contain any explicit API version (rename it from '{self.api_endpoint}' to just '{api}')"
            )
        )
        self.api_endpoint = api
```

Comment (David-Kristek) [reply src/genai/credentials.py:31]: DeprecationWarning is ignored by default (https://docs.python.org/3/library/exceptions.html#DeprecationWarning), should I rename it to Warning or is this wanted?

**Review by Tomas2D — APPROVED** [review]

**Pull request merged**